driver/driver.py

```diff
@@ -20,81 +9,266 @@
+    )
+
+    return mom_ensemble
+
+def configure_mom_ensemble( ensemble, args, colocated_orchestrator, db_nodes ):
+    """Configures all the members of the MOM6 ensemble with runtime configureable parameters
+    """
+
+    MOM6_config_options = {
+        "SIM_DAYS": 1, # length of simlations
+        "EKE_MODEL": args.eke_model_name,
+        "EKE_BACKEND": args.eke_backend,
+        "DOMAIN_LAYOUT": args.domain_layout,
+        "MASKTABLE": args.mask_table
     }
-for model in mom_ensemble:
-    model.params = MOM6_config
+    # For now the next entries have to be there to avoid the generator bug
+    MOM6_config_options.update( {
+        "SMARTREDIS_COLOCATED":"False",
+        "SMARTREDIS_COLOCATED_STRIDE":0,
+        "SMARTREDIS_CLUSTER": "False"
+    })
+
+    if colocated_orchestrator:
+        MOM6_config_options.update( {
+        "SMARTREDIS_COLOCATED":"True",
+        "SMARTREDIS_COLOCATED_STRIDE":18,
+        })
+    elif db_nodes >= 3:
+        MOM6_config_options.update( {'SMARTREDIS_CLUSTER':'True'} )
+
+    for model in ensemble:
+        model.params = MOM6_config_options
+        model.register_incoming_entity(model)
+
+    return ensemble
+
+def add_colocated_orchestrator( ensemble, args ):
+    for model in ensemble:
+        model.colocate_db(
+            args.orchestrator_port,
+            ifname=args.orchestrator_interface,
+            limit_app_cpus=False
+        )
+
+def create_distributed_orchestrator( args ):
+    orchestrator = Orchestrator(
+        port = args.orchestrator_port,
+        interface = args.orchestrator_interface,
+        db_nodes = args.orchestrator_nodes,
+        time="1:00:00",
+        threads_per_queue=4,
+        launcher='slurm',
+        batch=True)
+    orchestrator.set_cpus(18)
+    orchestrator.set_batch_arg("constraint", args.orchestrator_node_features)
+    orchestrator.set_batch_arg("exclusive",None)
+    return orchestrator
+
+def driver( ensemble_args, orchestrator_args, mom6_args):
+    log_to_file("./driver.log", log_level='developer')
+
+    experiment = Experiment("AI-EKE-MOM6", launcher="slurm")
+    mom_ensemble = create_mom_ensemble( experiment, ensemble_args )
+    configure_mom_ensemble(
+        mom_ensemble,
+        mom6_args,
+        orchestrator_args.colocated_orchestrator,
+        orchestrator_args.orchestrator_nodes
+    )
+
+    experiment_entities = [ mom_ensemble ]
+
+    if args.colocated_orchestrator:
+        add_colocated_orchestrator(mom_ensemble, orchestrator_args)
+    else:
+        orchestrator = create_distributed_orchestrator(orchestrator_args)
+        experiment_entities.append(orchestrator)
+
+    experiment.generate( *experiment_entities, overwrite=True )
+    experiment.start( *experiment_entities, summary=True )
+    print(experiment.summary())
+    experiment.stop()
+
+if __name__ == "__main__":
+    import argparse
+    parser = argparse.ArgumentParser(
```

Comment:
this is ALOT of arguments. consider using something like fire (which we use in the scaling tests). it will automatically create these descriptions and what not based on docstrings. 

It will also allow you to pass in dictionaries which will be useful in making this work accross systems.